src/api/general-services/pipeline-manage/index.js

```diff
@@ -0,0 +1,176 @@
+
+const crypto = require('crypto');
+const jq = require('jq-web');
+const YAML = require('yaml');
+const _ = require('lodash');
+const AWSXRay = require('aws-xray-sdk');
+const fetch = require('node-fetch');
+const AWS = require('../../../utils/requireAWS');
+const config = require('../../../config');
+const logger = require('../../../utils/logging');
+
+const getPipelineImages = async () => {
+  const response = await fetch(
+    config.pipelineInstanceConfigUrl,
+    {
+      method: 'GET',
+    },
+  );
+
+  const txt = await response.text();
+  const manifest = YAML.parseAllDocuments(txt);
+
+  return jq.json(manifest, '..|objects|.images//empty');
+};
+
+const getClusterInfo = async () => {
+  if (config.clusterEnv === 'development') return {};
+
+  const eks = new AWS.EKS({
+    region: config.awsRegion,
+  });
+
+  const { cluster: info } = await eks.describeCluster({ name: `biomage-${config.clusterEnv}` }).promise();
+  const { name, endpoint, certAuthority } = info;
+
+  return {
+    name,
+    endpoint,
+    certAuthority,
+  };
+};
+
+const constructPipelineStep = (context, step) => {
+  const { XStepType: stepType } = step;
```

Comment:
`XStepType` is a magic variable that determines which type of pipeline step we should plug in its place. Currently, there are two, `delete-completed-jobs`, which deletes completed jobs in k8s or locally, and `create-new-job-if-not-exist`, which creates a new remoter-server instance.